src/main/java/org/zalando/nakadi/controller/NakadiProblemHandling.java

```diff
@@ -0,0 +1,473 @@
+import org.zalando.nakadi.exceptions.runtime.InvalidLimitException;
+import org.zalando.nakadi.exceptions.runtime.InvalidPartitionKeyFieldsException;
+import org.zalando.nakadi.exceptions.runtime.InvalidStreamIdException;
+import org.zalando.nakadi.exceptions.runtime.InvalidVersionNumberException;
+import org.zalando.nakadi.exceptions.runtime.LimitReachedException;
+import org.zalando.nakadi.exceptions.runtime.NakadiBaseException;
+import org.zalando.nakadi.exceptions.runtime.NakadiRuntimeException;
+import org.zalando.nakadi.exceptions.runtime.NoStreamingSlotsAvailable;
+import org.zalando.nakadi.exceptions.runtime.NoSuchEventTypeException;
+import org.zalando.nakadi.exceptions.runtime.NoSuchPartitionStrategyException;
+import org.zalando.nakadi.exceptions.runtime.NoSuchSchemaException;
+import org.zalando.nakadi.exceptions.runtime.NoSuchStorageException;
+import org.zalando.nakadi.exceptions.runtime.NoSuchSubscriptionException;
+import org.zalando.nakadi.exceptions.runtime.NotFoundException;
+import org.zalando.nakadi.exceptions.runtime.PartitioningException;
+import org.zalando.nakadi.exceptions.runtime.RepositoryProblemException;
+import org.zalando.nakadi.exceptions.runtime.RequestInProgressException;
+import org.zalando.nakadi.exceptions.runtime.ServiceTemporarilyUnavailableException;
+import org.zalando.nakadi.exceptions.runtime.StorageIsUsedException;
+import org.zalando.nakadi.exceptions.runtime.TimeLagStatsTimeoutException;
+import org.zalando.nakadi.exceptions.runtime.TimelineException;
+import org.zalando.nakadi.exceptions.runtime.TooManyPartitionsException;
+import org.zalando.nakadi.exceptions.runtime.TopicCreationException;
+import org.zalando.nakadi.exceptions.runtime.UnableProcessException;
+import org.zalando.nakadi.exceptions.runtime.UnknownOperationException;
+import org.zalando.nakadi.exceptions.runtime.UnknownStorageTypeException;
+import org.zalando.nakadi.exceptions.runtime.UnprocessableEntityException;
+import org.zalando.nakadi.exceptions.runtime.UnprocessableSubscriptionException;
+import org.zalando.nakadi.exceptions.runtime.WrongInitialCursorsException;
+import org.zalando.nakadi.exceptions.runtime.WrongStreamParametersException;
+import org.zalando.problem.Problem;
+import org.zalando.problem.spring.web.advice.ProblemHandling;
+
+import static org.zalando.problem.Status.BAD_REQUEST;
+import static org.zalando.problem.Status.CONFLICT;
+import static org.zalando.problem.Status.FORBIDDEN;
+import static org.zalando.problem.Status.INTERNAL_SERVER_ERROR;
+import static org.zalando.problem.Status.NOT_FOUND;
+import static org.zalando.problem.Status.NOT_IMPLEMENTED;
+import static org.zalando.problem.Status.REQUEST_TIMEOUT;
+import static org.zalando.problem.Status.SERVICE_UNAVAILABLE;
+import static org.zalando.problem.Status.TOO_MANY_REQUESTS;
+import static org.zalando.problem.Status.UNPROCESSABLE_ENTITY;
+
+
+public interface NakadiProblemHandling extends ProblemHandling {
```

Comment:
why should we extend ProblemHandling? do we use everything from there? for example I see validation trait there, I remember `ValidationProblem.java` in EventTypeController.create(), which we create by ourselves, but in case we have validation trait I expect to be created for us.